puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/source/katello.groovy

```diff
@@ -94,13 +78,20 @@ pipeline {
                         }
                     }
                 }
-                stage('assets-precompile') {
+                stage('react-ui and assets-precompile') {
+                    // putting these together so npm install can be done in parallel and not block the other tests
                     steps {
                         dir('foreman') {
-                            withRVM(["bundle exec rake plugin:assets:precompile[${project_name}] RAILS_ENV=production --trace"], ruby)
-                        }
-                    }
-                }
+                            withRVM(["bundle exec npm install"], ruby)
+                        }   
+                        dir('katello') {
+                          sh 'npm test'
```

Comment:
There was an 'npm install' step here, but I removed it as foreman's npm install will install for plugins. I didn't see any installation happening in our current runs.

Reply:
Also, there was this condition:
```
when {
  expression { fileExists('package.json') }
}
```
Is this still needed? It seems implied now. If it is still needed, I could use some help on how to add it for just a single action in a step.